practice-app/app/views.py

```diff
@@ -242,3 +252,48 @@ def worldTime():
                 languageCode=world_time.languageCode,
             )
         return render_template("worldtime.html")
+
+@app.route("/pokemon", methods=["GET", "POST"])
+def PokemonPage():
+    if request.method == "GET":
+        return render_template("pokemon.html")
+    else:
+        pokemon_name = request.form.get("poke_name")
+        url = f"https://pokeapi.co/api/v2/pokemon/{pokemon_name.lower()}"
+
+        response = requests.get(url)
+        if response.status_code == 200:
+            data = response.json()
+            pokemon = Pokemon(
+                name=data["name"], height=data["height"], weight=data["weight"]
+            )
+            return render_template("pokemon.html", response=pokemon)
+        else:
+            return render_template(
+                "pokemon.html", error="Something went wrong. (Maybe wrong poke name) "
+            )
+
+
+@app.route(f"/pokesave", methods=["POST"])
+def AddPokeFavourites():
```

Comment:
Move to another file.

Use snake case, since this is a function - not a class name

Reply:
Moved to "pokemon_api.py". thanks for your suggestion.